backend/src/test/java/com/backend/backend/BookingControllerTest.java

```diff
@@ -0,0 +1,176 @@
+                .getContentAsString();
+
+        assertEquals(res, response);
+
+    }
+
+    @Test
+    public void testPostBookingSad1_NotFoundUser() throws Exception {
+        // Arrange
+        Booking booking = new Booking(
+                "674f56cca02d3de66bcaebcc",
+                "U456",
+                new Date(),
+                "Pending",
+                2
+        );
+
+        String bookingJson = objectMapper.writeValueAsString(booking);
+
+        String res = "Error: User not found!";
+
+        Mockito.when(bookingService.addBooking(Mockito.any(Booking.class))).thenReturn(res);
+
+        // Act
+        String response = mockMvc.perform(MockMvcRequestBuilders.post("/booking/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .content(bookingJson))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        assertEquals(res, response);
+
+    }
+
+    @Test
+    public void testPostBookingSad1_NotFoundDestination() throws Exception {
+        // Arrange
+        Booking booking = new Booking(
+                "674f56cca02d3de66bcaebcc",
+                "U456",
+                new Date(),
+                "Pending",
+                2
+        );
+
+        String bookingJson = objectMapper.writeValueAsString(booking);
+
+        String res = "Error: Destination not found!";
+
+        Mockito.when(bookingService.addBooking(Mockito.any(Booking.class))).thenReturn(res);
+
+        // Act
+        String response = mockMvc.perform(MockMvcRequestBuilders.post("/booking/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .content(bookingJson))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        assertEquals(res, response);
+
+    }
+
+    @Test
+    public void testPostBookingSad1_DuplicateBooking() throws Exception {
+        // Arrange
+        Booking booking = new Booking(
+                "674f56cca02d3de66bcaebcc",
+                "U456",
+                new Date(),
+                "Pending",
+                2
+        );
+
+        String bookingJson = objectMapper.writeValueAsString(booking);
+
+        String res = "Error: Duplicate Booking Record!";
+
+        Mockito.when(bookingService.addBooking(Mockito.any(Booking.class))).thenReturn(res);
+
+        // Act
+        String response = mockMvc.perform(MockMvcRequestBuilders.post("/booking/add")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .content(bookingJson))
+                .andReturn()
+                .getResponse()
+                .getContentAsString();
+
+        assertEquals(res, response);
+
+    }
+
+}
```

Comment:
The test class is well-structured with clear setup and comprehensive test cases, but it lacks parameterized tests, leading to potential redundancy.